### PR TITLE
go.mod: Use upstream gologin

### PR DESCRIPTION
**Pull request by indradhanush**
## Test plan

- [ ] main-dry-run should pass
- [ ] GitHub enterprise OAuth login should continue to work

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -504,7 +504,6 @@ replace (
 // =================================
 // These entries indicate replace directives that are defined for unknown reasons.
 replace (
-	github.com/dghubble/gologin => github.com/sourcegraph/gologin v1.0.2-0.20181110030308-c6f1b62954d8
```
Comment (unknwon) [inline go.mod:507]: You probably also need to change the line to a newer version that includes our patch commit:

https://github.com/sourcegraph/sourcegraph/blob/1f61168410743a2c63ecd81c9ae1243f5330f87e/go.mod#L37

Comment (indradhanush) [reply go.mod:507]: It was first available in [2.1.0](https://github.com/dghubble/gologin/releases/tag/v2.1.0). Nevertheless I still needed to update it based on the CI failure.

File: go.mod [hunk 2]
```diff
@@ -248,6 +247,7 @@ require (
 	github.com/c2h5oh/datasize v0.0.0-20220606134207-859f65c6625b
 	github.com/coreos/go-iptables v0.6.0
 	github.com/crewjam/saml/samlidp v0.0.0-20221211125903-d951aa2d145a
+	github.com/dghubble/gologin v2.1.0+incompatible
```
Comment (unknwon) [inline go.mod:250]: this looks like a downgrade? Is it intended?

**Comment by eseliger** [comment]
Closing as outdated.

**Pull request closed without merging**